plugins/woocommerce/src/Internal/Updates/BatchProcessingController.php

```diff
@@ -0,0 +1,200 @@
+	 *
+	 * @return int Action ID.
+	 */
+	private function schedule_next_batch( string $update_name ) : int {
+		return as_schedule_single_action( time(), self::SINGLE_PROCESS_MIGRATION_ACTION, array( $update_name ) );
+	}
+
+	/**
+	 * Whether the given updater is already scheduled.
+	 * Differs from `as_has_scheduled_action` in that this excludes actions in progress.
+	 *
+	 * @param string $update_name Fully qualified class name of the updator.
+	 *
+	 * @return bool Whether it's already scheduled.
+	 */
+	public function already_scheduled( string $update_name ) {
+		return as_has_scheduled_action( self::SINGLE_PROCESS_MIGRATION_ACTION, array( $update_name ) );
+	}
+
+	/**
+	 * Get instance of updater for given class name,
+	 *
+	 * @param string $updater_class_name Class name of updator.
+	 *
+	 * @return BatchProcessor Updator instance.
+	 */
+	private function get_processor_instance( string $updater_class_name ) : BatchProcessor {
+		$processor = wc_get_container()->get( $updater_class_name );
+
+		/**
+		 * Filters the instance of processor for current class name.
+		 *
+		 * @since 6.7.0.
+		 */
+		return apply_filters( 'woocommerce_get_batch_processor', $processor, $updater_class_name );
+	}
+
+	/**
+	 * Helper method to get list of all pending updates.
+	 *
+	 * @return array List of pending updates.
+	 */
+	public function get_pending() : array {
+		return get_option( self::PENDING_PROCESS_OPTION_NAME, array() );
+	}
+
+	/**
+	 * Mark a pending update complete.
+	 *
+	 * @param string $updater Name of updater class which will be marked compete.
+	 */
+	protected function mark_pending_process_complete( string $updater ) {
+		$pending_processes = $this->get_pending();
+		if ( in_array( $updater, $pending_processes ) ) {
+			$pending_processes = array_diff( $pending_processes, array( $updater ) );
+			$this->set_pending_processes( $pending_processes );
+		}
+	}
+
+	/**
+	 * Helper method set pending processes.
+	 *
+	 * @param array $processes List of pending processes.
+	 */
+	private function set_pending_processes( array $processes ) {
+		update_option( self::PENDING_PROCESS_OPTION_NAME, $processes, false );
+	}
+
+	/**
+	 * Start timestamp for first recurring action.
+	 *
+	 * @return int time internval.
+	 */
+	private function get_start_timestamp_for_init_action() {
+		return time() + MINUTE_IN_SECONDS;
+	}
+
+	/**
+	 * Arguments for recurring action.
+	 *
+	 * @return array
+	 */
+	private function get_args_for_init_cron() {
+		return array();
+	}
+
+	/**
+	 * Check if a particular process is pending.
+	 *
+	 * @param string $process_id Fully qualified class name of process.
+	 *
+	 * @return bool Whether the update is in progress.
+	 */
+	public function is_batch_process_pending( string $process_id ) : bool {
+		return in_array( $process_id, $this->get_pending() );
+	}
+
+	/**
+	 * Stops all pending updates, and clears current memory state.
+	 */
+	public function force_clear_all_processes() {
```

Comment:
Name change suggestion:

```suggestion
	public function cancel_queue() {
```

Reply:
So this does a bit more than canceling queue, as it will also clear the cron process that manages everything and not just the processors. That's I wanted this method to have bit more aggressive name.

Reply:
Ok, then maybe I would have gone for something more succint as `cancel_all`, but I won't argue you if you are willing to get aggressive. 😄

Reply:
😅, let's keep it as it is for now then.